controllers/external-accounts.js

```diff
@@ -1,10 +1,17 @@
 const externalAccountsModel = require("../models/external-accounts");
 
 const addExternalAccountData = async (req, res) => {
+  const createdOn = Date.now();
+
   try {
-    const createdOn = Date.now();
     const data = { ...req.body, createdOn };
 
+    // Check if token already exists
+    const dataFound = await externalAccountsModel.fetchExternalAccountData("", data.token);
```

Comment:
Hey @rohan09-raj if possible can you give a brief on what's happening from line 10-12

Reply:
We are generating a random token using `crpyto.subtle`, there are cases which people pointed that after certain time they experienced duplicate tokens. It's very very rare and the developers of that package resolved the issue too but just in case if we get a duplicate token generated. We should not be putting that request data in the database and throw `token already exists` message.